include/nbl/video/IGPUCommandPool.h

```diff
@@ -31,13 +66,219 @@ class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
         // Vulkan: const VkCommandPool*
         virtual const void* getNativeHandle() const = 0;
 
+        core::CMemoryPool<core::GeneralpurposeAddressAllocator<uint32_t>, core::default_aligned_allocator, false, uint32_t> m_commandSegmentPool;
+
     protected:
         virtual ~IGPUCommandPool() = default;
 
         core::bitflag<E_CREATE_FLAGS> m_flags;
         uint32_t m_familyIx;
 };
 
+class alignas(COMMAND_ALIGNMENT) IGPUCommandPool::ICommand
+{
+public:
+    virtual ~ICommand() {}
+
+    // static void* operator new(std::size_t size) = delete;
+    static void* operator new[](std::size_t size) = delete;
+    // static void* operator new(std::size_t size, std::align_val_t al) = delete;
+    static void* operator new[](std::size_t size, std::align_val_t al) = delete;
+
+    // static void operator delete  (void* ptr) = delete;
+    static void operator delete[](void* ptr) = delete;
+    static void operator delete  (void* ptr, std::align_val_t al) = delete;
+    static void operator delete[](void* ptr, std::align_val_t al) = delete;
+    static void operator delete  (void* ptr, std::size_t sz) = delete;
+    static void operator delete[](void* ptr, std::size_t sz) = delete;
+    static void operator delete  (void* ptr, std::size_t sz, std::align_val_t al) = delete;
+    static void operator delete[](void* ptr, std::size_t sz, std::align_val_t al) = delete;
+
+    uint32_t m_size;
```

Comment:
it shouldn't just dangle publicly, make it private, and add a public const getter.

 if you need someone to be able to set it, declare a friendship.

Reply:
hmm because we have a vtable anyway (the destructor) we could get rid of `m_size` and have a
```cpp
public:
    virtual uint32_t getSize() const = 0;
```

since the vtable pointer is basically like a typeID and we can deduce the length inside the derived class anyway.

Reply:
Are you saying that we should store `m_size` in derived classes instead?

Reply:
nah I was wrong, `m_size` in the base class is very useful for the iterator implementation.